File: UnityScripts/StoryModel.cs
using System;
using System.Collections.Generic;

[Serializable]
public class StoryModel
{
    public string id;
    public string requestor_name;
    public string topic;
    public List<Scenario> scenario;
}

[Serializable]
public class Scenario
{
    public string character;
    public string text;
    public string sound;
}

File: models/story_model.py
from datetime import datetime
from typing import List

from pydantic import BaseModel, Field


class Scenario(BaseModel):
    character: str
    text: str
    sound: str

class StoryModel(BaseModel):
    id: str = Field(..., alias='_id')
    topic_type: str
    requestor_name: str
    topic: str
    scenario: List[Scenario]
    created_at: datetime = Field(default_factory=datetime.utcnow)
